pvlib/spectrum/mismatch.py

```diff
@@ -571,3 +571,111 @@ def spectral_factor_caballero(precipitable_water, airmass_absolute, aod500,
     return modifier
+
+
+def spectral_factor_pvspec(airmass_absolute, clearsky_index,
+                           module_type=None, coefficients=None):
+    r"""
+    Estimate a technology-specific spectral mismatch modifier from absolute
+    airmass and clear sky index using the PVSPEC model.
+
+    The motivation for this model is to include the effects of cloud cover on
+    the spectrum, and thus spectral mismatch factor estimation. Another
+    motivation is to develop a simple parameterisation compared with existing
+    models. Model coefficients are derived using spectral irradiance and other
+    meteorological data from eight locations. These coefficients for seven
+    modules, available here via the ``module_type`` parameter, as well as more
+    details on the model, can be found in [1]_.
+
+    Parameters
+    ----------
+    airmass_absolute : numeric
+        absolute (pressure-adjusted) airmass. [unitless]
+
+    clearsky_index: numeric
+        clear sky index. [unitless]
+
+    module_type : str, optional
+        One of the following PV technology strings from [1]_:
+
+        * ``'fs4-1'`` - First Solar series 4-1 and earlier CdTe module.
+        * ``'fs4-2'`` - First Solar 4-2 and later CdTe module.
+        * ``'monosi'``, - anonymous sc-si module.
+        * ``'multisi'``, - anonymous mc-si- module.
+        * ``'cigs'`` - anonymous copper indium gallium selenide module.
+        * ``'asi'`` - anonymous amorphous silicon module.
+
+    coefficients : array-like, optional
+        user-defined coefficients, if not using one of the default coefficient
+        sets via the ``module_type`` parameter.
+
+    Returns
+    -------
+    mismatch: numeric
+        spectral mismatch factor (unitless) which is multiplied
+        with broadband irradiance reaching a module's cells to estimate
+        effective irradiance, i.e., the irradiance that is converted to
+        electrical current.
+
+    Notes
+    -----
+    The PVSPEC model parameterises the spectral mismatch factor as a function
+    of absolute air mass and the clear sky index as follows:
+
+    .. math::
+
+       M = a_1 K_c^{a_2} AM_a^{a_3},
+
+    where M is the spectral mismatch factor, and :math:`a_1, a_2, a_3` are
+    module-specific coefficients. In the PVSPEC model publication, absolute air
+    mass is estimated starting from the Kasten and Young relative air mass
+    [2]_. The clear sky index, which is the ratio of GHI to clear sky GHI, uses
+    the ESRA model [3]_ to estimate the clear sky GHI with monthly Linke
+    turbidity values from [4]_ as inputs.
+
+    References
+    ----------
+    .. [1] Pelland, S., Beswick, C., Thevenard, D., Côté, A., Pai, A. and
+       Poissant, Y., 2020, June. Development and testing of the PVSPEC model of
+       photovoltaic spectral mismatch factor. In 2020 47th IEEE Photovoltaic
+       Specialists Conference (PVSC) (pp. 1258-1264). IEEE.
+       :doi:`10.1109/PVSC45281.2020.9300932`
```

Comment:
Something weird happens with this reference when you build the docs... maybe @kandersolar or @AdamRJensen can help?

Reply:
What is the weird thing that happens?

Reply:
This is how it looks when you open the readthedocs:

![image](https://github.com/pvlib/pvlib-python/assets/88548539/f5316c8c-4567-4f30-bf88-5b45237524d0)

Reply:
You mean the (1,2) ? That is just because there are two references to this paper. If you click the 1 or the 2 in the brackets, it takes you to the first or the second reference to the paper in the text above.

Reply:
Yep, that's just how the docs output looks.  We could look into changing it (I don't see much value in the (1, 2) and it confuses everyone at first), but that should definitely be a separate PR.  Nothing to fix here :)

Reply:
Yeah can't lie it threw me too at first, then I realised it was just a hyperlink to the in-text references